### PR TITLE
executor: fix column_default with timestamp and bit type

### LINKED ISSUE #36027: In `information_schema`.`columns` table, timestamp column's column_default is inconsistent with MySQL
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

> mysql> create table t ( `b` timestamp(3) NOT NULL DEFAULT '1970-01-01 08:00:01.000' COMMENT '');
> Query OK, 0 rows affected (0.08 sec)
> 
> mysql> show create table t;
> +-------+----------------------------------------------------------------------------------------------------------------------------------------------+
> | Table | Create Table                                                                                                                                 |
> +-------+----------------------------------------------------------------------------------------------------------------------------------------------+
> | t     | CREATE TABLE `t` (
>   `b` timestamp(3) NOT NULL DEFAULT '1970-01-01 08:00:01.000'
> ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
> +-------+----------------------------------------------------------------------------------------------------------------------------------------------+
> 1 row in set (0.00 sec)
> 
> mysql> select column_default from information_schema.columns where TABLE_NAME='t';
> +-------------------------+
> | column_default          |
> +-------------------------+
> | 1970-01-01 00:00:01.000 |
> +-------------------------+
> 1 row in set (0.03 sec)

### 2. What did you expect to see? (Required)

In mysql:

> mysql> create table t ( `b` timestamp(3) NOT NULL DEFAULT '1970-01-01 08:00:01.000' COMMENT '');
> Query OK, 0 rows affected (0.00 sec)
> 
> mysql> show create table t;
> +-------+-------------------------------------------------------------------------------------------------------------------------+
> | Table | Create Table                                                                                                            |
> +-------+-------------------------------------------------------------------------------------------------------------------------+
> | t     | CREATE TABLE `t` (
>   `b` timestamp(3) NOT NULL DEFAULT '1970-01-01 08:00:01.000'
> ) ENGINE=InnoDB DEFAULT CHARSET=latin1 |
> +-------+-------------------------------------------------------------------------------------------------------------------------+
> 1 row in set (0.00 sec)
> 
> mysql> select column_default from information_schema.columns where TABLE_NAME='t';
> +-------------------------+
> | column_default          |
> +-------------------------+
> | 1970-01-01 08:00:01.000 |
> +-------------------------+
> 1 row in set (0.01 sec)

### 3. What did you see instead (Required)

show create table's result and information_schema.columns's result are  inconsistent!

### 4. What is your TiDB version? (Required)

> mysql> mysql> select version();
> +-----------------------------------------+
> | version()                               |
> +-----------------------------------------+
> | 5.7.25-TiDB-v5.4.0-alpha-178-g02a8f0453 |
> +-----------------------------------------+
> 1 row in set (0.00 sec)

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: close #36027

Problem Summary: 
fix two bugs:
1. The value of `column_default` not consider time_zone, see #36027
2. `show create table` for `create table t(a bit default (rand()))` not correct

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/infoschema_reader.go [hunk 1]
```diff
@@ -891,10 +891,19 @@ ForColumnsTag:
...
+				if ft.GetType() == mysql.TypeTimestamp && columnDefault != types.ZeroDatetimeStr {
+					timeValue, err := table.GetColDefaultValue(sctx, col)
+					if err == nil {
+						columnDefault = timeValue.GetMysqlTime().String()
+					}
+				}
+				if ft.GetType() == mysql.TypeBit {
+					defaultValBinaryLiteral := types.BinaryLiteral(columnDefault.(string))
+					columnDefault = defaultValBinaryLiteral.ToBitLiteralString(true)
+				}
```
Comment (tangenta) [inline executor/infoschema_reader.go:906]: Perhaps this should be tuned to `DefaultExpr` as well:

```sql
mysql> create table t(a bit default (rand()));
Query OK, 0 rows affected (0.05 sec)

mysql> select column_default from information_schema.columns where TABLE_NAME='t' and TABLE_SCHEMA='test';
+----------------+
| COLUMN_DEFAULT |
+----------------+
| rand()         |
+----------------+
1 row in set (0.02 sec)
```

Comment (Defined2014) [reply executor/infoschema_reader.go]: forgot it, fixed. PTAL
@tangenta

**Review by tangenta — APPROVED** [review]

File: executor/show.go [hunk 2]
```diff
@@ -1013,7 +1013,7 @@ func ConstructResultOfShowCreateTable(ctx sessionctx.Context, tableInfo *model.T
 						defaultValStr = timeValue.GetMysqlTime().String()
 					}
 
-					if col.GetType() == mysql.TypeBit {
+					if col.GetType() == mysql.TypeBit && !col.DefaultIsExpr {
```
Comment (xhebox) [inline executor/show.go:1016]: What about:

```
if col.DefaultIsExpr {
} else if col.GetType() == mysql.TypeBit {
}
```

**Review by xhebox — COMMENTED** [review]
Rest LGTM

File: executor/show.go (hunk 2 above)
Comment (Defined2014) [reply executor/show.go]: Done @xhebox

**Review by xhebox — APPROVED** [review]

**Pull request merged**